cd /media/LUN1/NIlesh_analysis/Exome/mutsigcv/
for f in *.*; do sed -i "s/$/\t$f/" $f; done

cat *.txt >> mutsigcv_input.maf



sed -i '1iHugo_Symbol\tEntrez_Gene_Id\tCenter\tNCBI_Build\tChromosome\tStart_position\tEnd_position\tStrand\tVariant_Classification\tVariant_Type\tReference_Allele\tTumor_Seq_Allele1\tTumor_Seq_Allele2\tdbSNP_RS\tdbSNP_Val_Status\tTumor_Sample_Barcode\tMatched_Norm_Sample_Barcode\tMatch_Norm_Seq_Allele1\tMatch_Norm_Seq_Allele2\tTumor_Validation_Allele1\tTumor_Validation_Allele2\tMatch_Norm_Validation_Allele1\tMatch_Norm_Validation_Allele2\tVerification_Status\tValidation_Status\tMutation_Status\tSequencing_Phase\tSequence_Source\tValidation_Method\tScore\tBAM_file\tSequencer\tTumor_Sample_UUID\tMatched_Norm_Sample_UUID\tGenome_Change\tAnnotation_Transcript\tTranscript_Strand\tTranscript_Exon\tTranscript_Position\tcDNA_Change\tCodon_Change\tProtein_Change\tOther_Transcripts\tRefseq_mRNA_Id\tRefseq_prot_Id\tSwissProt_acc_Id\tSwissProt_entry_Id\tDescription\tUniProt_AApos\tUniProt_Region\tUniProt_Site\tUniProt_Natural_Variations\tUniProt_Experimental_Info\tGO_Biological_Process\tGO_Cellular_Component\tGO_Molecular_Function\tCOSMIC_overlapping_mutations\tCOSMIC_fusion_genes\tCOSMIC_tissue_types_affected\tCOSMIC_total_alterations_in_gene\tTumorscape_Amplification_Peaks\tTumorscape_Deletion_Peaks\tTCGAscape_Amplification_Peaks\tTCGAscape_Deletion_Peaks\tDrugBank\tref_context\tgc_content\tCCLE_ONCOMAP_overlapping_mutations\tCCLE_ONCOMAP_total_mutations_in_gene\tCGC_Mutation_Type\tCGC_Translocation_Partner\tCGC_Tumor_Types_Somatic\tCGC_Tumor_Types_Germline\tCGC_Other_Diseases\tDNARepairGenes_Role\tFamilialCancerDatabase_Syndromes\tMUTSIG_Published_Results\tOREGANNO_ID\tOREGANNO_Values\t1000Genome_AA\t1000Genome_AC\t1000Genome_AF\t1000Genome_AFR_AF\t1000Genome_AMR_AF\t1000Genome_AN\t1000Genome_ASN_AF\t1000Genome_AVGPOST\t1000Genome_CIEND\t1000Genome_CIPOS\t1000Genome_END\t1000Genome_ERATE\t1000Genome_EUR_AF\t1000Genome_HOMLEN\t1000Genome_HOMSEQ\t1000Genome_LDAF\t1000Genome_RSQ\t1000Genome_SNPSOURCE\t1000Genome_SVLEN\t1000Genome_SVTYPE\t1000Genome_THETA\t1000Genome_VT\tACHILLES_Lineage_Results_Top_Genes\tCGC_Cancer_Germline_Mut\tCGC_Cancer_Molecular_Genetics\tCGC_Cancer_Somatic_Mut\tCGC_Cancer_Syndrome\tCGC_Chr\tCGC_Chr_Band\tCGC_GeneID\tCGC_Name\tCGC_Other_Germline_Mut\tCGC_Tissue_Type\tCOSMIC_n_overlapping_mutations\tCOSMIC_overlapping_mutation_descriptions\tCOSMIC_overlapping_primary_sites\tClinVar_ASSEMBLY\tClinVar_HGMD_ID\tClinVar_SYM\tClinVar_TYPE\tClinVar_rs\tCosmic\tESP_AA\tESP_AAC\tESP_AA_AC\tESP_AA_AGE\tESP_AA_GTC\tESP_AvgAAsampleReadDepth\tESP_AvgEAsampleReadDepth\tESP_AvgSampleReadDepth\tESP_CA\tESP_CDP\tESP_CG\tESP_CP\tESP_Chromosome\tESP_DBSNP\tESP_DP\tESP_EA_AC\tESP_EA_AGE\tESP_EA_GTC\tESP_EXOME_CHIP\tESP_FG\tESP_GL\tESP_GM\tESP_GS\tESP_GTC\tESP_GTS\tESP_GWAS_PUBMED\tESP_MAF\tESP_PH\tESP_PP\tESP_Position\tESP_TAC\tESP_TotalAAsamplesCovered\tESP_TotalEAsamplesCovered\tESP_TotalSamplesCovered\tEnsembl_so_accession\tEnsembl_so_term\tFamilial_Cancer_Genes_Reference\tFamilial_Cancer_Genes_Synonym\tHGNC_Ensembl_Gene_ID\tHGNC_HGNC_ID\tHGNC_RefSeq_IDs\tHGNC_Status\tHGNC_UCSC_ID(supplied_by_UCSC)\tHGVS_coding_DNA_change\tHGVS_genomic_change\tHGVS_protein_change\tMyLabDB\tORegAnno_bin\tSource\tUniProt_alt_uniprot_accessions\tannotation_transcript\tbuild\tccds_id\tdbNSFP_1000Gp1_AC\tdbNSFP_1000Gp1_AF\tdbNSFP_1000Gp1_AFR_AC\tdbNSFP_1000Gp1_AFR_AF\tdbNSFP_1000Gp1_AMR_AC\tdbNSFP_1000Gp1_AMR_AF\tdbNSFP_1000Gp1_ASN_AC\tdbNSFP_1000Gp1_ASN_AF\tdbNSFP_1000Gp1_EUR_AC\tdbNSFP_1000Gp1_EUR_AF\tdbNSFP_Ancestral_allele\tdbNSFP_CADD_phred\tdbNSFP_CADD_raw\tdbNSFP_CADD_raw_rankscore\tdbNSFP_ESP6500_AA_AF\tdbNSFP_ESP6500_EA_AF\tdbNSFP_Ensembl_geneid\tdbNSFP_Ensembl_transcriptid\tdbNSFP_FATHMM_pred\tdbNSFP_FATHMM_rankscore\tdbNSFP_FATHMM_score\tdbNSFP_GERP++_NR\tdbNSFP_GERP++_RS\tdbNSFP_GERP++_RS_rankscore\tdbNSFP_Interpro_domain\tdbNSFP_LRT_Omega\tdbNSFP_LRT_converted_rankscore\tdbNSFP_LRT_pred\tdbNSFP_LRT_score\tdbNSFP_LR_pred\tdbNSFP_LR_rankscore\tdbNSFP_LR_score\tdbNSFP_MutationAssessor_pred\tdbNSFP_MutationAssessor_rankscore\tdbNSFP_MutationAssessor_score\tdbNSFP_MutationTaster_converted_rankscore\tdbNSFP_MutationTaster_pred\tdbNSFP_MutationTaster_score\tdbNSFP_Polyphen2_HDIV_pred\tdbNSFP_Polyphen2_HDIV_rankscore\tdbNSFP_Polyphen2_HDIV_score\tdbNSFP_Polyphen2_HVAR_pred\tdbNSFP_Polyphen2_HVAR_rankscore\tdbNSFP_Polyphen2_HVAR_score\tdbNSFP_RadialSVM_pred\tdbNSFP_RadialSVM_rankscore\tdbNSFP_RadialSVM_score\tdbNSFP_Reliability_index\tdbNSFP_SIFT_converted_rankscore\tdbNSFP_SIFT_pred\tdbNSFP_SIFT_score\tdbNSFP_SLR_test_statistic\tdbNSFP_SiPhy_29way_logOdds\tdbNSFP_SiPhy_29way_logOdds_rankscore\tdbNSFP_SiPhy_29way_pi\tdbNSFP_UniSNP_ids\tdbNSFP_Uniprot_aapos\tdbNSFP_Uniprot_acc\tdbNSFP_Uniprot_id\tdbNSFP_aaalt\tdbNSFP_aapos\tdbNSFP_aapos_FATHMM\tdbNSFP_aapos_SIFT\tdbNSFP_aaref\tdbNSFP_cds_strand\tdbNSFP_codonpos\tdbNSFP_fold-degenerate\tdbNSFP_genename\tdbNSFP_hg18_pos(1-coor)\tdbNSFP_phastCons100way_vertebrate\tdbNSFP_phastCons100way_vertebrate_rankscore\tdbNSFP_phastCons46way_placental\tdbNSFP_phastCons46way_placental_rankscore\tdbNSFP_phastCons46way_primate\tdbNSFP_phastCons46way_primate_rankscore\tdbNSFP_phyloP100way_vertebrate\tdbNSFP_phyloP100way_vertebrate_rankscore\tdbNSFP_phyloP46way_placental\tdbNSFP_phyloP46way_placental_rankscore\tdbNSFP_phyloP46way_primate\tdbNSFP_phyloP46way_primate_rankscore\tdbNSFP_refcodon\tdbSNP\tgencode_transcript_name\tgencode_transcript_status\tgencode_transcript_tags\tgencode_transcript_type\tgene_id\tgene_type\thavana_transcript\tsecondary_variant_classification\tstrand\tsample' /media/LUN4/prajish/exome/processed_data/mutsigcv/mutsigcv_input.maf

cut -f 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,35,36,37,38,39,40,41,42,267 mutsigcv_input.maf >> mutsigcv_input_corrected.maf

#sed -i '1iHugo_Symbol\tEntrez_Gene_Id\tCenter\tNCBI_Build\tChromosome\tStart_position\tEnd_position\tStrand\tVariant_Classification\tVariant_Type\tReference_Allele\tTumor_Seq_Allele1\tTumor_Seq_Allele2\tdbSNP_RS\tdbSNP_Val_Status\tTumor_Sample_Barcode\tMatched_Norm_Sample_Barcode\tMatch_Norm_Seq_Allele1\tMatch_Norm_Seq_Allele2\tTumor_Validation_Allele1\tTumor_Validation_Allele2\tMatch_Norm_Validation_Allele1\tMatch_Norm_Validation_Allele2\tVerification_Status\tValidation_Status\tMutation_Status\tSequencing_Phase\tSequence_Source\tValidation_Method\tScore\tBAM_file\tSequencer\tGenome_Change\tAnnotation_Transcript\tTranscript_Strand\tTranscript_Exon\tTranscript_Position\tcDNA_Change\tCodon_Change\tProtein_Change\tis_coding\tis_silent\tcateg' /media/LUN1/NIlesh_analysis/mutsigcv/20151903/mutsigcv_input_corrected.maf

# Download mutsigcv_input_corrected.maf file and paste sample name column below Tumor bacode
sed -i 'liHugo_Symbol\tEntrez_Gene_Id\tCenter\tNCBI_Build\tChromosome\tStart_position\tEnd_position\tStrand\tVariant_Classification\t  Variant_Type\tReference_Allele\tTumor_Seq_Allele1\tTumor_Seq_Allele2\tdbSNP_RS\tdbSNP_Val_Status Tumor_Sample_Barcode\t Matched_Norm_Sample_Barcode\tMatch_Norm_Seq_Allele1\tMatch_Norm_Seq_Allele2\tTumor_Validation_Allele1\tTumor_Validation_Allele2\t    Match_Norm_Validation_Allele1\tMatch_Norm_Validation_Allele2\tVerification_Status\tValidation_Status\tMutation_Status\tSequencing_Phase\t  Sequence_Source\tValidation_Method Score\tBAM_file\tSequencer\tGenome_Change\tAnnotation_Transcript\tTranscript_Strand\t     Transcript_Exon\tTranscript_Position\tcDNA_Change\tCodon_Change\tProtein_Change' /media/LUN2/bikram/mutsigCV/corrected_mutsigCV.maf
 


awk '{print $1,"\t",$2,"\t",$3,"\t",$4,"\t",$5,"\t",$6,"\t",$7,"\t",$8,"\t",$9,"\t",$10,"\t",$11,"\t",$12,"\t",$13,"\t",$14,"\t",$15,"\t",$16,"\t",$17,"\t",$18,"\t",$19,"\t",$20,"\t",$21,"\t",$22,"\t",$23,"\t",$24,"\t",$25,"\t",$26,"\t",$27,"\t",$28,"\t",$29,"\t",$30,"\t",$31,"\t",$32,"\t",$35,"\t",$36,"\t",$37,"\t",$38,"\t",$39,"\t",$40,"\t",$41,"\t",$42}' 
echo "Mutsig started at $(date)"
bash /opt/ngs_softwares/MutSigCV/MutSigCV_1.4/run_MutSigCV.sh /usr/local/MATLAB/MATLAB_Compiler_Runtime/v81/ /media/LUN4/prajish/exome/processed_data/mutsigcv/mutsigcv_input_corrected.maf /opt/ngs_softwares/MutSigCV/references/exome_full192.coverage.txt /opt/ngs_softwares/MutSigCV/references/gene.covariates.txt /media/LUN4/prajish/exome/processed_data/mutsigcv/Output.txt /opt/ngs_softwares/MutSigCV/references/mutation_type_dictionary_file.txt /opt/ngs_softwares/MutSigCV/references/chr_files_hg19/
echo "Mutsig completed at $(date)"

awk '{print $1,"\t",$2,"\t",$3,"\t",$4,"\t",$5,"\t",$6,"\t",$7,"\t",$8,"\t",$9,"\t",$10,"\t",$11,"\t",$12,"\t",$13,"\t",$14,"\t",$15,"\t",$16=FILENAME"T","\t",$17=FILENAME"N","\t",$18,"\t",$19,"\t",$20,"\t",$21,"\t",$22,"\t",$23,"\t",$24,"\t",$25,"\t",$26,"\t",$27,"\t",$28,"\t",$29,"\t",$30,"\t",$31,"\t",$32,"\t",$35,"\t",$36,"\t",$37,"\t",$38,"\t",$39,"\t",$40,"\t",$41,"\t",$42}' "/media/LUN2/bikram/mutsigCV/1_cor.maf" | cut -f 16 | sed 's|.*/||' | cut -f 17 | sed 's|.*/||' >> "/media/LUN2/bikram/mutsigCV/1_cor_new.maf"
